File: src/main/java/de/pbauerochse/worklogviewer/fx/Theme.java
package de.pbauerochse.worklogviewer.fx;

/**
 * Available themes for the WorklogViewer UI
 */
public enum Theme {

    /**
     * The light theme as it has always been
     * present in the worklog viewer
     */
    LIGHT("/fx/css/light.css"),

    /**
     * A new darker theme
     */
    DARK("/fx/css/dark.css");

    private final String stylesheet;

    Theme(String stylesheet) {
        this.stylesheet = stylesheet;
    }

    public String getStylesheet() {
        return stylesheet;
    }
}

File: application/src/main/java/de/pbauerochse/worklogviewer/fx/Theme.kt
package de.pbauerochse.worklogviewer.fx

import de.pbauerochse.worklogviewer.util.FormattingUtil.getFormatted

/**
 * Available themes for the WorklogViewer UI
 */
enum class Theme(val stylesheet: String, val webviewStylesheet : String) {

    /**
     * The light theme as it has always been
     * present in the worklog viewer
     */
    LIGHT("/fx/css/light.css", "/fx/css/webview.light.css"),

    /**
     * A new darker theme
     */
    DARK("/fx/css/dark.css", "/fx/css/webview.dark.css");

    override fun toString(): String {
        return getFormatted("theme.${name.toLowerCase()}.label")
    }
}
